Backport PR #12830 to 7.x: ecs-compatibility docs (#13016)
elastic/logstash: docs/static/ecs-compatibility.asciidoc

If left unspecified for an individual plugin, the pipeline's `pipeline.ecs_compatibility` setting will be observed.
This allows you to configure plugins to use ECS -- or to lock in your existing non-ECS behavior -- in advance of your {ls} 8.0 upgrade where ECS will be enabled by default.

ECS Compatibility modes do not prevent you from explicitly configuring a plugin in a manner that conflicts with ECS.
Instead, they ensure that _implicit_ configuration avoids conflicts.

NOTE: Until {ls} 8.0 and the final 7.x are released, any value for `pipeline.ecs_compatibility` other than `disabled` are considered BETA and unsupported.
      As we continue to release plugins with ECS Compatibility modes, having this flag set will cause even patch-level upgrades to _automatically_ consume breaking changes in the upgraded plugins, changing the shape of data the plugin produces.

ifeval::["{ls8-ecs-major-version}"!="v8"]
NOTE: ECS `v8` will be the default in the GA release of {ls} 8.0.0, and will be available at or before the final minor release of {ls} 7.

>       We expect the scope of breaking changes in ECS 8 to be limited, but progress toward the definition of ECS v8 can be tracked https://github.com/elastic/ecs/issues/839[here].

Reviewer: ```suggestion
      We expect the scope of breaking changes in ECS 8 to be limited.
      We are https://github.com/elastic/ecs/issues/839[tracking progress toward ECS v8] in a GitHub issue.
```

Reviewer: Using "here" as link text is not a best practice.  Instead, make the link text as informative and relevant as possible. 

**SEO:** Search engines prioritize links over other text, and "here" doesn't provide any valuable information to improve our rankings. 
**ADA compliance.** If a user asks a screen reader for a list of links, "here" isn't helpful. 

I missed this in the first review.  :-(
